src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -160,10 +160,16 @@ export class CellModel extends Disposable implements ICellModel {
 			if (!this._attachments) {
 				this._attachments = {};
 			}
-			// TO DO: Check if name already exists and message the user?
-			this._attachments[name] = attachment;
-			this.sendChangeToNotebook(NotebookChangeType.CellMetadataUpdated);
+			// Check if name already exists and get a unique name
+			if (this._attachments[name] && this._attachments[name][mimeType] !== attachment[mimeType]) {
```

Comment:
It seems like if the mimeType is different that should be an error or you should still give it a unique name - because right now that case would result in the "new" one being ignored which could result in the wrong thing being referenced.

Reply:
am not sure which case is being ignored, if the mimeType is different the second condition will be true and it would still get a unique name right.

Reply:
If the name exists in the array already but the mime type is different then you won't attach anything (the second check is only for names that don't already exist)